plugins/woocommerce/src/Blocks/BlockTypes/AbstractInnerBlock.php

```diff
@@ -14,12 +14,53 @@ abstract class AbstractInnerBlock extends AbstractBlock {
 	protected $is_lazy_loaded = true;
 
 	/**
-	 * Registers the block type with WordPress using the metadata file.
+	 * Registers the block type with WordPress.
 	 *
-	 * The registration using metadata is now recommended. And it's required for "Inner Blocks" to
-	 * fix the issue of missing translations in the inspector (in the Editor mode)
+	 * This method determines whether to use compiled metadata or file-based metadata
+	 * for block registration.
 	 */
 	protected function register_block_type() {
+		$this->load_compiled_block_metadata();
+
+		if ( isset( parent::$compiled_block_metadata[ $this->namespace . '/' . $this->block_name ] ) ) {
+			$this->register_block_type_from_compiled_metadata();
+		} else {
+			$this->register_block_type_from_file_metadata();
```

Comment:
I think this change makes sense if we don't have a `watch:build` command, though I think I'd slightly lean towards have a watch command so that we can mimic the build as closely as possible and help catch bugs earlier.

Reply:
I was mostly thinking of developers before it was in the watch build (or developers who didn't know about it). If we do end up going w/ this PR, it could be removed